handlers/guest_links_test.go

```diff
@@ -347,3 +346,164 @@ func TestDeleteInvalidGuestLink(t *testing.T) {
+				ID:             picoshare.GuestLinkID("abcdefgh23456789"),
+				Created:        mustParseTime("2024-01-01T00:00:00Z"),
+				UrlExpires:     mustParseExpirationTime("2030-01-02T03:04:25Z"),
+				FileLifetime:   picoshare.FileLifetimeInfinite,
+				MaxFileBytes:   picoshare.GuestUploadUnlimitedFileSize,
+				MaxFileUploads: picoshare.GuestUploadUnlimitedFileUploads,
+				IsDisabled:     true,
+			},
+			status: http.StatusNoContent,
+		},
+		{
+			description: "enabling guest link succeeds but has no effect when guest link is not enabled",
+			url:         "/api/guest-links/abcdefgh23456789/enable",
+			guestLinkInStore: picoshare.GuestLink{
+				ID:             picoshare.GuestLinkID("abcdefgh23456789"),
+				Created:        mustParseTime("2024-01-01T00:00:00Z"),
+				UrlExpires:     mustParseExpirationTime("2030-01-02T03:04:25Z"),
+				FileLifetime:   picoshare.FileLifetimeInfinite,
+				MaxFileBytes:   picoshare.GuestUploadUnlimitedFileSize,
+				MaxFileUploads: picoshare.GuestUploadUnlimitedFileUploads,
+				IsDisabled:     false,
+			},
+			expected: picoshare.GuestLink{
+				ID:             picoshare.GuestLinkID("abcdefgh23456789"),
+				Created:        mustParseTime("2024-01-01T00:00:00Z"),
+				UrlExpires:     mustParseExpirationTime("2030-01-02T03:04:25Z"),
+				FileLifetime:   picoshare.FileLifetimeInfinite,
+				MaxFileBytes:   picoshare.GuestUploadUnlimitedFileSize,
+				MaxFileUploads: picoshare.GuestUploadUnlimitedFileUploads,
+				IsDisabled:     false,
+			},
+			status: http.StatusNoContent,
+		},
+		{
+			description: "disable a nonexistent guest link",
+			url:         "/api/guest-links/abcdefgh23456789/disable",
+			status:      http.StatusNotFound,
+		}, {
+			description: "enable a nonexistent guest link",
+			url:         "/api/guest-links/abcdefgh23456789/enable",
+			status:      http.StatusNotFound,
+		},
+		{
+			description: "disable a invalid guest link",
+			url:         "/api/guest-links/i-am-an-invalid-link/disable",
+			status:      http.StatusBadRequest,
+		},
+		{
+			description: "enable a invalid guest link",
+			url:         "/api/guest-links/i-am-an-invalid-link/enable",
+			status:      http.StatusBadRequest,
+		},
+	} {
+		t.Run(tt.description, func(t *testing.T) {
+			dataStore := test_sqlite.New()
+
+			if tt.status == http.StatusNoContent {
+				if err := dataStore.InsertGuestLink(tt.guestLinkInStore); err != nil {
+					t.Fatalf("failed to insert dummy guest link: %v", err)
+				}
+			}
+
+			s := handlers.New(mockAuthenticator{}, &dataStore, nilSpaceChecker, nilGarbageCollector, handlers.NewClock())
+
+			req, err := http.NewRequest("PUT", tt.url, nil)
+			if err != nil {
+				t.Fatalf("failed to create request for url: %s, error: %v", tt.url, err)
+			}
+			rec := httptest.NewRecorder()
+			s.Router().ServeHTTP(rec, req)
+			res := rec.Result()
+
+			if got, want := res.StatusCode, tt.status; got != want {
+				t.Errorf("%s: handler returned wrong status code: got %v want %v",
```

Comment:
This should be `t.Fatalf`. If the status code is wrong, there's no sense in checking the response data.

And can we simplify to this to match the other tests?

```go
if got, want := res.StatusCode, tt.status; got != want {
  t.Fatalf("status=%d, want=%d", got, want)
}
```

Including the description is an artifact from before I realized we could do `t.Run`.